data-access-layer/iaas/AliClient.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const Storage = require('ali-oss');
+// const Compute = require('@alicloud/pop-core');
+const logger = require('../../common/logger');
+// const errors = require('../../common/errors');
+// const utils = require('../../common/utils');
+// const uuid = require('uuid');
+// const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+// const NotFound = errors.NotFound;
+// const Unauthorized = errors.Unauthorized;
+// const Forbidden = errors.Forbidden;
+
+class AliClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    // this.constructor.validateParams(_.chain(this.settings).value());
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+    logger.debug('Looking for container ' + container);
+    return Promise.try(() => {
+      return this.storage.listBuckets({
+        prefix: container
+      })
+        .then(buckets => {
+          if (buckets.buckets === null) {
+            logger.error('Bucket ' + container + ' does not exists');
+          } else if (buckets.buckets.length == 1) {
+            return buckets.buckets;
+            logger.info('Bucket ' + container + ' exists');
+          } else {
+            logger.error('More than 1 Buckets with prefix ' + container + ' exists');
+          }
+        })
+        .catch(err => {
+          logger.error('Bucket ' + container + ' does not exists');
+        });
+    });
+  }
+
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    return this.storage
+      .useBucket(container)
+      .list(options)
+      .then(listOfFiles => {
+        let list = listOfFiles.objects;
+        let isTruncated = listOfFiles.isTruncated;
```

Comment:
Please check if these are applicable for Ali. Remove code which is AWS specific.

Reply:
Verified this part. The logic is similar to AWS, but indeed works for Ali too.